index.js

```diff
@@ -616,7 +618,7 @@ class Batch {
   async put (key, value, opts) {
     const release = this.batchLock ? await this.batchLock() : null
 
-    const cas = (opts && opts.cas) || null
+    const cas = (opts && opts.cas) || (this.tree.onlyIfChanged ? sameData : null)
```

Comment:
That's  a tricky one in terms of API. What's the intended behaviour if a user specified `onlyIfChanged: true`, but also passes an explicit cas? 

ATM it will ignore `onlyIfChanged` if the cas is passed. The alternative would be to run both the cas and the `onlyIfChanged`, and to proceed only if both pass. Not saying that is better, but it's important to explicitly decide what the behaviour should be.

Also in terms of API: I would also allow passing `onlyIfChanged` as an option to `put` and `del`

Reply:
Yeah, I think the intended behavior is that `cas`  overwrites `onlyIfChanged`.

Reply:
What about this?
https://github.com/holepunchto/hbd/blob/aecebed30ad64cd3838a9d2b2d2d70f4ad7443a3/index.js#LL87C7-L87C19

Reply:
API is ok because method option overrides constructor option, just like with Hypercore timeout options, etc